plugins/backend/catalog-backend-module-okta/src/providers/OktaGroupEntityProvider.ts

```diff
@@ -103,13 +110,25 @@ export class OktaGroupEntityProvider extends OktaEntityProvider {
           this.logger.warn(`failed to add user to group: ${e.message}`);
         }
       });
-
+      const profileAnnotations: Record<string, string> = {};
+      if (this.customAttributesToAnnotationWhitelist.length) {
+        for (const [key, value] of new Map(Object.entries(group.profile))) {
```

Comment:
I wish `GroupEntity['spec']['profile']` was more flexible. I wonder what the tradeoffs are between using labels vs using annotations? Thoughts?

Reply:
To me it makes sense to be an annotation. I think this is exactly what annotations are used for to store some additional data in a key-value pairs that you'll later work on, base some conditional rendering or using them in processors.